static/app/views/dashboards/widgets/tableWidget/tableWidgetVisualization.spec.tsx

```diff
@@ -0,0 +1,67 @@
+import {render, screen} from 'sentry-test/reactTestingLibrary';
+
+import {DurationUnit, RateUnit} from 'sentry/utils/discover/fields';
+import type {TabularData} from 'sentry/views/dashboards/widgets/common/types';
+import {sampleHTTPRequestTableData} from 'sentry/views/dashboards/widgets/tableWidget/fixtures/sampleTableData';
+import {TabularColumnFixture} from 'sentry/views/dashboards/widgets/tableWidget/fixtures/tabularColumn';
+import {TableWidgetVisualization} from 'sentry/views/dashboards/widgets/tableWidget/tableWidgetVisualization';
+
+describe('tableWidgetVisualization', function () {
+  it('Basic table renders correctly', async function () {
+    render(<TableWidgetVisualization tableData={sampleHTTPRequestTableData} />);
+
+    expect(await screen.findByText('http.request_method')).toBeInTheDocument();
+    expect(await screen.findByText('count(span.duration)')).toBeInTheDocument();
+  });
+
+  it('Table applies columns prop and their custom names over fallback if provided', async function () {
+    render(
+      <TableWidgetVisualization
+        tableData={sampleHTTPRequestTableData}
+        columns={TabularColumnFixture()}
+      />
+    );
+
+    expect(await screen.findByText('http request_method')).toBeInTheDocument();
+    expect(await screen.findByText('count span.duration')).toBeInTheDocument();
```

Comment:
Another good assertion to add would be to check that the column order matches what's specified in `TabularColumnFixture()`